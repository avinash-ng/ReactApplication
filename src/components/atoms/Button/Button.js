import React from 'react'
import PropTypes from 'prop-types'
const Button = (props ) => {
    return (
        <button className="a_button" className={props.className} onClick={props.onClicked}>{props.buttonName}</button>
    )
}

Button.propTypes = {
    buttonName: PropTypes.string.isRequired
}

export default Button
