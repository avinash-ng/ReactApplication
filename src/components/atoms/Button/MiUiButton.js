import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton