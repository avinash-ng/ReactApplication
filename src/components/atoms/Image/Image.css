img {
    width: 450px;
    height: 450px;
    float: left;
    margin-left: 20px;
    margin-top: 0px;
    padding-top: 60px;
    /* margin-bottom: 30px; */
}