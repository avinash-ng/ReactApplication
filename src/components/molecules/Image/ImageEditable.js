import React from 'react'
import ImageTag from '../../atoms/ImageTag/ImageTag'
import Button from "../../atoms/Button/Button"
const ImageEditable = props => {
    return (
        <div className="tag-div" contentEditable="true" suppressContentEditableWarning={true}>
            {/* <h3 className = "h3-image-tag" style={{ width: '450px' }}  onClick={() => changeHeading(each, index)} >{each.name}</h3> */}
            <ImageTag className={props.className} onClicked={props.headingOnClicked} object={props.object}></ImageTag>
            {props.object.selected ? <Button buttonName={props.buttonName} onClicked={props.buttonOnClicked}></Button> : ""}

        </div>
    )
}
export default ImageEditable