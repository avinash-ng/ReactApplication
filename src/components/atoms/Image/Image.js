import React from 'react'
import PropTypes from "prop-types"
import "./Image.css"
const Image = (props) => {
    return (
        <img data-testid="image" src={props.imageName} alt={props.altText} />
    )
}
Image.propTypes = {
    imageName: PropTypes.string.isRequired,
    altText: PropTypes.string.isRequired
} 

export default Image