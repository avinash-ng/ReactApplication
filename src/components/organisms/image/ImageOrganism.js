import React from 'react'
import Image from '../../atoms/Image/Image'
import ImageEditable from '../../molecules/Image/ImageEditable'
const ImageOrganism = (props) => {
    return (
        <div className={props.divClassName}>
            <Image altText={props.imgAltText} imageName={"http://localhost:3001/uploads/"+props.object.name}></Image>
            <br />
            <ImageEditable className={props.imgClassName} headingOnClicked={props.headingOnClicked} object={props.object} buttonName={props.buttonName} buttonOnClicked={props.buttonOnClicked}></ImageEditable>
        </div>
    )
}

export default ImageOrganism