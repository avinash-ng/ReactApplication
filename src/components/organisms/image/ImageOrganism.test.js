const { shallow, configure, mount } = require("enzyme")
import ImageOrganism from "./ImageOrganism"
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import Button from "../../atoms/Button/Button";
import sinon from 'sinon'
import { render, fireEvent } from "@testing-library/react";
import Image from "../../atoms/Image/Image";
import ImageEditable from "../../molecules/Image/ImageEditable";
const object = {
    'name': 'max.png',
    'selected':true
}
const object1 = {
    'name': 'max.png',
    'selected':false
}
const alt = 'Error in loading....'
const imgClassName = 'image'
const buttonName= 'click em'
configure({ adapter: new Adapter() });
describe('testing <ImageOrganism /> ', () => {
    it('renders <Image /> correctly', () => {
        console.log(object)
        const RenderComponent = shallow(<ImageOrganism buttonName={buttonName} object={object} imgAltText={alt}> </ImageOrganism>)
        expect(RenderComponent.find(Image).exists()).toBe(true)
    })

    it('renders <ImageEditable /> correctly', () => {
        const RenderComponent = shallow(<ImageOrganism buttonName={buttonName} object={object} imgAltText={alt}> </ImageOrganism>)
        expect(RenderComponent.find(ImageEditable).exists()).toBe(true)
    })

    it('checking on click functionality ', () => {
        const deleteFunction = sinon.spy()
        const RenderComponent = mount(<ImageOrganism buttonName={buttonName} object={object} imgAltText={alt} buttonOnClicked={deleteFunction}> </ImageOrganism>)

        RenderComponent.find(Button).simulate('click')

        expect(deleteFunction).toHaveProperty('callCount', 1)


    })

    it('matches snapshot', () => {
        const RenderComponent = shallow(<ImageOrganism buttonName={buttonName}  object={object} imgAltText={alt}></ImageOrganism>)
        expect(RenderComponent).toMatchSnapshot()
    })
})