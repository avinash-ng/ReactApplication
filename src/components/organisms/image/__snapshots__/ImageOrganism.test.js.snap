// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <ImageOrganism />  matches snapshot 1`] = `ShallowWrapper {}`;
